extensions-contrib/druid-jwt-extensions/src/main/java/org/apache/druid/security/jwt/IssuerExtractor.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.security.jwt;
+
+import org.jose4j.jwt.MalformedClaimException;
+import org.jose4j.jwt.consumer.InvalidJwtException;
+import org.jose4j.jwt.consumer.JwtConsumer;
+import org.jose4j.jwt.consumer.JwtConsumerBuilder;
+
+/**
+ * Deocde JWT to extract issuer field in the JTW claims
+ */
+public class IssuerExtractor
+{
+  private final JwtConsumer baseConsumer;
+
+  IssuerExtractor()
+  {
+    baseConsumer = new JwtConsumerBuilder()
+        .setSkipAllValidators()
+        .setDisableRequireSignature()
+        .setSkipSignatureVerification()
+        .build();
```

Comment:
why do we disable signature validation on the JWT? As part of authentication we must validate the signature and enforce it.

Reply:
this jwt consumer extracts issuer only so that we can map jwt to the correct jwt instance. 

For example, 
```
druid.auth.authenticatorChain=["jwt-okta", "jwt-vault "]
```
in `do_filter`, we extract the issuer to filter the auth chain

Reply:
makes sense, a comment is probably warranted to explain that and mention that this method doesn't actually validate the JWT at all